File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Drink.java
package com.logistics.Components.Product;

public class Drink extends Consumable {

    private boolean isCarbonated;

    public Drink() {
        super();
        isCarbonated = false;
    }

    public Drink(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, int calories, boolean isCarbonated) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit, calories);
        this.isCarbonated = isCarbonated;
    }

    // Copy Constructor
    public Drink(Drink other) {
        super(other);
        this.isCarbonated = other.isCarbonated;
    }

    // Clone method
    @Override
    public Drink clone() {
        return new Drink(this);
    }

    // Getters
    public boolean getIsCarbonated() { return isCarbonated; }

    // Setters
    public void setIsCarbonated(boolean isCarbonated) { this.isCarbonated = isCarbonated; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nCarbonated: " + isCarbonated;
    }
}

File: Task 1/C++/Components/Product/Drink.cpp
#include "Drink.h"

Drink::Drink() : Consumable(){
    this->isCarbonated = false;
}

Drink::Drink(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, int calories, bool isCarbonated)
    : Consumable(name, cost, supplier, discount_amount, packing_units, cost_per_unit, calories){
    this->isCarbonated = isCarbonated;
}

Drink::Drink(const Drink& other) : Consumable(other) {
    this->isCarbonated = other.isCarbonated;
}

std::string Drink::display_product_info(){
    return Consumable::display_product_info() + 
            "\nIs Carbonated: " + (isCarbonated ? "True" : "False");
}